lib/circleci/circleci.go

```diff
@@ -45,6 +47,8 @@ func issuerURL(template, organizationID string) string {
 // ID token.
 // See https://circleci.com/docs/openid-connect-tokens/
 type IDTokenClaims struct {
+	oidc.Claims
```

Comment:
Iirc - if you have a `sub` field on the custom struct, you need to implement GetSubject() for Zitadel OIDC to be able to check the subject properly as the custom struct will take priority during unmarshaling.